app/controllers/api_controller/provision_requests.rb

```diff
@@ -18,5 +18,25 @@ def create_resource_provision_requests(type, _id, data)
       MiqProvisionVirtWorkflow.from_ws(version_str, @auth_user_obj, template_fields, vm_fields, requester, tags,
                                        additional_values, ems_custom_attrs, miq_custom_attrs)
     end
+
+    def deny_resource_provision_requests(type, id, data)
+      provreq = resource_search(id, type, collection_class(:provision_requests))
+      provreq.deny(@auth_user, data['reason'])
+      res = action_result(true, "Provision request #{id} denied")
+      add_href_to_result(res, type, id)
```

Comment:
Not all action responses include an href I believe, thus the method above.  For cookie cutter actions that require it, there is a common method api_action that does that generically, see the vms.rb action's, i.e. start_resource_vms, stop_resource_vms using api_action.  

If you want to use api_action here that's fine. Otherwise, I'm good with this PR :+1: 
Let me know, Thanks.